pkg/registry/service/rest_test.go

```diff
@@ -689,7 +689,7 @@ func TestServiceRegistryIPUpdate(t *testing.T) {
 
 	update = deepCloneService(created_service)
 	update.Spec.Ports[0].Port = 6503
-	update.Spec.ClusterIP = "1.2.3.76" // error
+	update.Spec.ClusterIP = "1.2.4.76" // Error: Provided Cluster IP is not in the range of 1.2.3.0/24
```

Comment:
@wojtek-t  Yes, you are right. 
It seems the IP Range is not the root reason. I have added the log and find the cluster IP is random generated.

**First time:**
<code>rest_test.go:690: &api.Service{TypeMeta:api.TypeMeta{Kind:"", APIVersion:""}, ObjectMeta:api.ObjectMeta{Name:"foo", GenerateName:"", Namespace:"default", SelfLink:"", UID:"9d07d5bb-3f8f-11e5-86d7-0242ac110035", ResourceVersion:"1", Generation:0, CreationTimestamp:util.Time{Time:time.Time{sec:63574828975, nsec:207762404, loc:(*time.Location)(0x11db7c0)}}, DeletionTimestamp:(*util.Time)(nil), Labels:map[string]string(nil), Annotations:map[string]string(nil)}, Spec:api.ServiceSpec{Ports:[]api.ServicePort{api.ServicePort{Name:"", Protocol:"TCP", Port:6503, TargetPort:util.IntOrString{Kind:0, IntVal:6502, StrVal:""}, NodePort:0}}, Selector:map[string]string{"bar":"baz"}, ClusterIP:"1.2.3.31", Type:"ClusterIP", DeprecatedPublicIPs:[]string(nil), SessionAffinity:"None"}, Status:api.ServiceStatus{LoadBalancer:api.LoadBalancerStatus{Ingress:[]api.LoadBalancerIngress(nil)}}}</code>

**Second time**
<code>rest_test.go:690: &api.Service{TypeMeta:api.TypeMeta{Kind:"", APIVersion:""}, ObjectMeta:api.ObjectMeta{Name:"foo", GenerateName:"", Namespace:"default", SelfLink:"", UID:"7b46ca93-3f95-11e5-8764-0242ac110037", ResourceVersion:"1", Generation:0, CreationTimestamp:util.Time{Time:time.Time{sec:63574831495, nsec:558187561, loc:(*time.Location)(0x11db7c0)}}, DeletionTimestamp:(*util.Time)(nil), Labels:map[string]string(nil), Annotations:map[string]string(nil)}, Spec:api.ServiceSpec{Ports:[]api.ServicePort{api.ServicePort{Name:"", Protocol:"TCP", Port:6503, TargetPort:util.IntOrString{Kind:0, IntVal:6502, StrVal:""}, NodePort:0}}, Selector:map[string]string{"bar":"baz"}, ClusterIP:"1.2.3.239", Type:"ClusterIP", DeprecatedPublicIPs:[]string(nil), SessionAffinity:"None"}, Status:api.ServiceStatus{LoadBalancer:api.LoadBalancerStatus{Ingress:[]api.LoadBalancerIngress(nil)}}}</code>

The Cluster IP is created randomly while calling this function:
<code>r := ipallocator.NewCIDRRange(makeIPNet(t))</code>
https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/registry/service/ipallocator/allocator.go#L87
https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/registry/service/allocator/bitmap.go#L61

So it happens to me that he cluster IP is generated as 1.2.3.76 and the test case failed. I rollback the code and testcase never fails again.

This issue may cause the build failed randomly, maybe also influence the CI, Shippable. 
Can you help to make sure of that? I can help to fix this issue. Thanks.